src/view/widgets/multi_select_widget.py

```diff
@@ -97,25 +83,207 @@ def create(self, config: dict[str, Any]) -> pn.widgets.MultiSelect | None:
+        linked_filter_name = config.get(c.MULTI_SELECT_LINKED_FILTER_NAME_KEY)
+        filter_category = config.get(c.MULTI_SELECT_FILTER_CATEGORY)
+
+        if (
+            (upper_item_filter_widget is None)
+            or (not linked_filter_name)
+            or (not filter_category)
+        ):
+            return
+
+        upper_item_filter_widget.param.watch(
+            lambda event: self.load_options_after_filter_got_set(
+                linked_filter_name, filter_category, upper_item_filter_widget, event
+            ),
+            "value",
+            onlychanged=True,
+        )
+        self.controller_instance.register("upper_item_filter", upper_item_filter_widget)
+        return upper_item_filter_widget
+
+    def load_options_after_filter_got_set(
+        self, target_widget_name, category, filter_widget, event
+    ):
+        """
+        Loads options for item filter after upper item filter was triggered. First select the referenced item filter by name
+        and then get and insert new filter options for the target item filter based on the selected upper item filter.
+
+        :param target_widget_name: target item filter widgets name.
+        :param category: category to set how to get new option values for target item filter.
+        :param filter_widget: upper item filter widget.
+        :param event: event type triggered by upper item filter.
+        """
+        if (event.type != "changed") or (category != "genres"):
+            return
+
+        target_widget = self.find_widget_by_name_recursive(
+            self.reco_explorer_app_instance.config_based_nav_controls,
+            target_widget_name,
+        )
+
+        target_widget.value = (
+            self.controller_instance.get_genres_and_subgenres_from_upper_category(
+                filter_widget.value, category
             )
-            return model_choice_widget.create(config)
-        else:
-            return self.build_multi_select_widget(config)
+        )
+
+    def find_widget_by_name_recursive(self, widget, target_name):
+        """
+        Check if a widget has the target name. Function is recursive and calls itself while iterating over all
+        widgets and widgets nested in that widgets.
+
+        :param widget: a panel widget object
+        :param target_name: the name to check the widget for
+        """
+        # Check if the current widget has the target name
+        if hasattr(widget, "name") and widget.name == target_name:
+            return widget
+
+        # If the current widget has nested widgets, iterate over them
+        if hasattr(widget, "objects"):
+            for obj in widget.objects:
+                # Recursively call the function to search within nested widgets
+                found_widget = self.find_widget_by_name_recursive(obj, target_name)
+                if found_widget is not None:
+                    return found_widget
+
+        return None
```

Comment:
Der Gedanke hinter dem zusammenfassen hier ist, dass die Filter auch nur veränderte Multi Selects drin sind und die Klassen ein bisschen "ping pong" spielen? (Multi Select ruft Filter auf, Filter ruft aber auch wieder was aus Mutli Select auf...) Aber wird die dann hier nicht doch etwas sehr groß? (fast 300 Zeilen) Oder steckt noch was dahinter, das ich nicht direkt sehe?

Reply:
Genau. In der ursprünglichen Implementierung wurde das Multiselect an die Filter übergeben. Um da die Build-Methode aufzurufen.
Wie du schon sagst konzeptionell sind die Filter ja immer noch multiselects und brauchen auch immer die Methoden davon.
Deswegen hab ich mir gedacht wenn man statt das multiselect direkt zu übergeben, einfach die Methoden erbt passt das ein bisschen besser von der Abstraktion.

Wenn du magst kann ich für die Übersicht die Filter gerne wieder in eigene Files auslagern, das ist kein Problem.

Reply:
Ich habs mir noch mal genauer angeguckt. Passt so glaub ich für mich. Länge des Files ist noch vertretbar, "einen Tod muss man sterben" ;)